### DISCUSSION TITLE
How to bring an object to the front?

### DISCUSSION THREAD

**Discussion by luiz00martins** [Pyroxenium/Basalt#29] [Q&A] [answered]
Say you add an object A to the frame, and then an Object B, in that order. It seems that, by default, B will be in front of A.

How may I dynamically bring A or B to the front of the frame?

**Comment by NoryiE** [comment, marked as answer]
When you create 2 objects of the same type, they have by default the same z-index (for example buttons have 5). This means the latest object will be the most foreground object if it's the same z-index. However you can change the z-index by using for example :setZIndex(6)

https://basalt.madefor.cc/#/objects/Object/setZIndex

**Reply by luiz00martins** [reply]
Yep, that seems to work, thank you!

I seem to have found a minor bug though. The object seems to forget some settings after changing the Z-Index:
```lua
-- This will yield a stylized button.
local input = main_frame:addInput("amount_popup")
	:setZIndex(10) -- note that this comes first
	:setPosition(1, 1)
	:setSize(math.floor(width/2), 1)
	:setBackground(colors.black)
	:setForeground(colors.lightGray)
	:setDefaultText("Amount", colors.gray)
	:show()

-- This will yield a default button.
local input = main_frame:addInput("amount_popup")
	:setPosition(1, 2)
	:setSize(math.floor(width/2), 1)
	:setBackground(colors.black)
	:setForeground(colors.lightGray)
	:setDefaultText("Amount", colors.gray)
	:setZIndex(10) -- note that this comes last
	:show()
```

Should I open an issue?

**Reply by NoryiE** [reply]
Hello, thank you very much. I've fixed the issue in the dev branch, i think this should work now.
